### PR TITLE
feat: create a bundled build of wasm

### PR DESCRIPTION
**Pull request by jeswr**
Follows up on the discussion https://github.com/rla/npm-swipl-wasm/pull/8 to produce a `swipl-bundle.js` build which inlines the `.wasm` and `.data` files.

### CODE REVIEW

File: cmake/EmscriptenTargets.cmake [hunk 1]
```diff
@@ -39,20 +39,33 @@ set(WASM_WEB_LINK_FLAGS
...
+    -s SINGLE_FILE
+    --embed-file ${WASM_PRELOAD_DIR}@swipl)
+join_list(WASM_BUNDLE_LINK_FLAGS_STRING " " ${WASM_BUNDLE_LINK_FLAGS} " " ${WASM_SHARED_LINK_FLAGS})
+add_executable(swipl-bundle ${SWIPL_SRC})
+set_target_properties(swipl-bundle PROPERTIES
+		      LINK_FLAGS "${WASM_BUNDLE_LINK_FLAGS_STRING}")
+target_link_libraries(swipl-bundle libswipl)
+add_dependencies(swipl-bundle wasm_preload)
+set_property(TARGET swipl-bundle PROPERTY LINK_DEPENDS
+	     ${POSTJS} ${PREJS})
```
Comment (jeswr) [inline cmake/EmscriptenTargets.cmake:71]:
```suggestion
set(WASM_BUNDLE_EMBEDDED_LINK_FLAGS
    -s SINGLE_FILE
    --embed-file ${WASM_PRELOAD_DIR}@swipl)
set(WASM_BUNDLE_LINK_FLAGS
    -s SINGLE_FILE)
join_list(WASM_BUNDLE_LINK_FLAGS_STRING " " ${WASM_BUNDLE_LINK_FLAGS} " " ${WASM_SHARED_LINK_FLAGS})
join_list(WASM_BUNDLE_EMBEDDED_LINK_FLAGS_STRING " " ${WASM_BUNDLE_EMBEDDED_LINK_FLAGS} " " ${WASM_SHARED_LINK_FLAGS})
add_executable(swipl-bundle ${SWIPL_SRC})
set_target_properties(swipl-bundle PROPERTIES
		      LINK_FLAGS "${WASM_BUNDLE_LINK_FLAGS_STRING}")
target_link_libraries(swipl-bundle libswipl)
add_dependencies(swipl-bundle wasm_preload)
set_property(TARGET swipl-bundle PROPERTY LINK_DEPENDS
	     ${POSTJS} ${PREJS})

add_executable(swipl-bundle-embedded ${SWIPL_SRC})
set_target_properties(swipl-bundle-embedded PROPERTIES
		      LINK_FLAGS "${WASM_BUNDLE_EMBEDDED_LINK_FLAGS_STRING}")
target_link_libraries(swipl-bundle-embedded libswipl)
add_dependencies(swipl-bundle-embedded wasm_preload)
set_property(TARGET swipl-bundle-embedded PROPERTY LINK_DEPENDS
	     ${POSTJS} ${PREJS})
	    
```

**Comment by jeswr** [comment]
@JanWielemaker - This is ready for review

**Comment by JanWielemaker** [comment]
> This is ready for review

Hope to take care of this shortly.   I've a bit of a backlog on issues though :cry:

**Comment by JanWielemaker** [comment]
The bundle builds fine, but trying to use it in shell.html results in

    TypeError: Module.preRun is undefined

What does this change to how it is supposed to be used?

**Comment by jeswr** [comment]
`preRun` is used to perform operations before initializing `SWIPL` (for instance here https://github.com/eyereasoner/eye-js/blob/0cc5e11f44a042f322c6e77e3370d8b6b5d30011/scripts/generate-pvm.ts#L36-L39 it loads `eye.pl` before `swipl -q -f eye.pl` is run).

It should be able to be left undefined and in those cases just do nothing. I'm unsure why this is a new error - I don't think it would be caused by the new configuration file. In particular I did not get any errors of the sorts when using the output of this build file on commit b0896169e0c097bdea168937af36bc376f57252d (i.e. the commit that npm-swipl-wasm is using in the [DockerFile](https://github.com/rla/npm-swipl-wasm/blob/master/docker/Dockerfile)). Have you made any changes to any other code since then that may be the culprit?

**Comment by jeswr** [comment]
I tried having a quick look at [this diff](https://github.com/SWI-Prolog/swipl-devel/compare/master..b0896169e0c097bdea168937af36bc376f57252d) but could identify anything that might be the cause on first parse.

**Comment by jeswr** [comment]
I've tested these changes on the latest version of swipl-devel in [this branch](https://github.com/rla/npm-swipl-wasm/compare/master...jeswr:npm-swipl-wasm:chore/testing?expand=1). The `npm-swipl-wasm` tests run fine after the changes in #1102 are made.

**Pull request closed without merging**